roachtest: upload GEOS libs for sqlsmith

otan opened the pull request:
* Add code to recognise GEOS code locations.
* Upload GEOS related files into any roachtest cluster.
* Configured builds to save lib*/libgeos* artifacts in
  [MergeToMaster](https://teamcity.cockroachdb.com/admin/editBuild.html?id=buildType:Cockroach_MergeToMaster)
  and take them from
  [Nightlies](https://teamcity.cockroachdb.com/admin/editDependencies.html?id=buildType:Cockroach_Nightlies_WorkloadNightly).
* Make sqlsmith explicitly upload geospatial related directories.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/51377)
<!-- Reviewable:end -->

madelynnblue reviewed (approved):
LGTM with caveat that I'm not that qualified to comment about roachtest (I didn't add the sqlsmith roachtest).

otan committed:
roachtest: upload GEOS libs for sqlsmith

* Add code to recognise GEOS code locations.
* Upload GEOS related files into any roachtest cluster.
* Configured builds to save lib*/libgeos* artifacts in
  [MergeToMaster](https://teamcity.cockroachdb.com/admin/editBuild.html?id=buildType:Cockroach_MergeToMaster)
  and take them from
  [Nightlies](https://teamcity.cockroachdb.com/admin/editDependencies.html?id=buildType:Cockroach_Nightlies_WorkloadNightly).
* Make sqlsmith explicitly upload geospatial related directories.

Release note: None

otan commented:
bors r=mjibson
